### DISCUSSION TITLE
Can we log raw query that prisma is using under the hood by using `prisma.$use()` api.

### DISCUSSION THREAD

**Discussion by ashish336b** [prisma/prisma#17261] [Q&A] [answered]
I want to log raw SQL query Prisma is using under the hood to retrieve data. I think using middleware is best suite for my case prisma.$use(). I could not find how to implement this functionality.

**Comment by nurul3101** [comment, marked as answer]
Hello @ashish336b 👋 

You would need to [configure logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) to get the Raw SQL query that Prisma is executing under the hood.

**Reply by ashish336b** [reply]
Thanks @nurul3101

**Reply by nurul3101** [reply]
No worries! :)

**Reply by revero-doug** [reply]
how should we go about parsing `params` on the `query` type? they're not properly serialized json

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
